onnxoptimizer/passes/fuse_concat_and_reshape.h

```diff
@@ -41,7 +42,13 @@ struct FuseConcatAndReshape final : public PredicateBasedPass {
       } else if (graph.is_constant_initializer(v)) {
         tensor = &*graph.getInitializer(v->uniqueName());
       } else {
-        if (v->sizes().size() != 1) {
+        // If the value of v is unknown, and v has only one element, we can represent
+        // it with -1 in Reshape op.
+        // TODO:
+        //  support the case that v is a shape and only one of the dims is unknown.
+        //  Example:
+        //  Concat [?, 2, 3] and [4] into [-1, 2, 3, 4]
+        if (v->sizes().size() != 1 || !v->sizes()[0].is_int || v->sizes()[0].dim != 1) {
```

Comment:
fix the bug:

`v->sizes().size() != 1` should be `v->sizes()[0].dim != 1`

Reply:
yes，you are right.  `v->sizes().size() != 1`  only check the shape of v is `1D`, but`1D` maybe consist of not only one element.